frontend/src/actions/session_actions.js

```diff
@@ -20,13 +20,31 @@ export const receiveErrors = errors => ({
 export const login = user => dispatch =>
   ApiUtil.login(user).then(
     userP => {
-      dispatch(receiveCurrentUser(userP.data.user));
+      dispatch(receiveCurrentUser(userP.data));
+      return userP;
     },
     err => {
       dispatch(receiveErrors(err.response.data));
+      return err.response.data;
     }
   );
 
+export const sendMail = email => dispatch =>
+  ApiUtil.sendMail(email).then((res) => {
+      return res;
+    }
+  )
+
+export const sessionLogin = user => dispatch =>
+  ApiUtil.sessionLogin(user).then(
```

Comment:
try to use the async/await syntax instead of the promise-chaining syntax wherever possible. It is a lot more concise/readable